File: src/main/java/example/micronaut/BreakEvenRequestHandler.java
package example.micronaut;
import io.micronaut.core.annotation.Introspected;
import io.micronaut.function.aws.MicronautRequestHandler;
import java.util.UUID;

@Introspected
public class BreakEvenRequestHandler extends MicronautRequestHandler<BreakEvenRequest, BreakEvenResponse> {  // <1>

    @Override
    public BreakEvenResponse execute(BreakEvenRequest request) {
        BreakEvenResponse breakEvenResponse = new BreakEvenResponse();
        breakEvenResponse.breakEvenPoint = (int) Math.ceil(request.fixedCosts / (request.price - request.unitCosts));
        return breakEvenResponse;
    }
}

File: src/main/kotlin/info/novatec/BreakEvenRequestHandler.kt
package info.novatec
import io.micronaut.core.annotation.Introspected
import io.micronaut.function.aws.MicronautRequestHandler

@Introspected
class BreakEvenRequestHandler : MicronautRequestHandler<BreakEvenRequest?, BreakEvenResponse?>() {

    override fun execute(input: BreakEvenRequest?): BreakEvenResponse? {
        return if (input != null) {
            val response = BreakEvenResponse()
            response.breakEvenPoint = kotlin.math.ceil(input.fixedCosts / (input.price - input.unitCosts)).toInt()
            return response
        } else {
            null
        }
    }
}